cmd/kpod/import.go

```diff
@@ -0,0 +1,192 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/projectatomic/libpod/libpod"
+
+	"github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+var (
+	importFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "change, c",
+			Usage: "Apply imgspecv1 configurations to the created image",
+		},
+		cli.StringFlag{
+			Name:  "message, m",
+			Usage: "Set commit message for the image imported",
+		},
+	}
+	importDescription = `Imports a tarball and saves it as a root filesystem image.
+	 The commit message and image config can be modified by the user.`
+	importCommand = cli.Command{
+		Name:        "import",
+		Usage:       "Import a tarball to create a filesystem image",
+		Description: importDescription,
+		Flags:       importFlags,
+		Action:      importCmd,
+		ArgsUsage:   "TARBALL [REFERENCE]",
+	}
+)
+
+func importCmd(c *cli.Context) error {
+	if err := validateFlags(c, historyFlags); err != nil {
+		return err
+	}
+
+	runtime, err := getRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "Could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	var opts libpod.CopyOptions
+	var source string
+	args := c.Args()
+	switch len(args) {
+	case 0:
+		return errors.Errorf("need to give the path to the tarball")
+	case 1:
+		source = args[0]
+	case 2:
+		source = args[0]
+		opts.Reference = args[1]
+	default:
+		return errors.Errorf("too many arguments, need 2 only")
+	}
+
+	// if source is a url, download it and save to a temp file
+	u, err := url.ParseRequestURI(source)
+	if err == nil && u.Scheme != "" {
+		file, err := downloadFromURL(source)
+		if err != nil {
+			return err
+		}
+		defer os.Remove(file)
+		source = file
+	}
+
+	changes := v1.ImageConfig{}
+	if c.IsSet("change") {
```

Comment:
Change is a slice, should be a for loop.

Reply:
fixed